org.eclipse.scanning.api/src/org/eclipse/scanning/api/points/AbstractPosition.java

```diff
@@ -110,8 +116,17 @@ public final boolean equals(Object obj, boolean checkStep) {
 		final Map<String, Integer> iours   = getIndices();
 		final Map<String, Integer> itheirs = getIndices((IPosition)obj);
 		if (!iours.equals(itheirs)) return false;		
-	
-		if (!equals(getDimensionNames(), getDimensionNames((IPosition)obj))) return false;		
+
+		List<List<String>> ourDimNames = getDimensionNames().stream()
+				.map(c -> new ArrayList<String>(c))
+				.collect(Collectors.toList());
+		ourDimNames.forEach(c -> Collections.sort(c));
+		List<List<String>> theirDimNames = getDimensionNames((IPosition) obj).stream()
+				.map(c -> new ArrayList<String>(c))
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1612)